packages/core/src/sort.ts

```diff
@@ -100,42 +102,71 @@ function sortByDeviation(rows: Intersections, sortByOrder?: SortByOrder) {
   const { values, order } = rows;
   const newOrder = [...order].sort(
     (a, b) => {
-      const devA = values[a].deviation;
-      const devB = values[b].deviation;
+      const devA = values[a].attributes.deviation;
+      const devB = values[b].attributes.deviation;
       return (sortByOrder === 'Ascending') ? devA - devB : devB - devA;
     },
   );
 
   return { values, order: newOrder };
 }
 
+/**
+ * @param rows - The intersections object containing values and order.
+ * @param sortBy - The attribute to sort by
+ * @param sortByOrder - The sort order ('Ascending' or 'Descending'). Defaults to 'Ascending'.
+ * @returns The sorted intersections object.
+ */
+function sortByAttribute(rows: Intersections, sortBy: string, sortByOrder?: SortByOrder) {
+  const { values, order } = rows;
+
+  const newOrder = [...order].sort(
+    (a, b) => {
+      const meanA = (values[a].attributes[sortBy] as FiveNumberSummary).mean;
+      const meanB = (values[b].attributes[sortBy] as FiveNumberSummary).mean;
+
+      // If one of the values is undefined (empty subset), sort it to the bottom
+      if (!meanA) {
+        return 1;
+      }
+      if (!meanB) {
+        return -1;
+      }
+
+      return (sortByOrder === 'Ascending') ? meanA - meanB : meanB - meanA;
+    },
+  );
+
+  return { values, order: newOrder };
+}
+
 /**
  * Sorts the intersections based on the specified criteria.
  *
  * @template T - The type of intersections.
- * @param {T} intersection - The intersection to be sorted.
+ * @param {T} intersection - The intersections to be sorted.
  * @param {SortBy} sortBy - The criteria to sort the intersection by.
  * @param {SortVisibleBy} vSetSortBy - The criteria to sort the visible sets by.
  * @param {Sets} visibleSets - The visible sets.
  * @param {SortByOrder} [sortByOrder] - The order in which to sort the intersection.
  * @returns {T} The sorted intersection.
  */
 function sortIntersections<T extends Intersections>(
-  intersection: T,
+  intersections: T,
   sortBy: SortBy,
   vSetSortBy: SortVisibleBy,
   visibleSets: Sets,
   sortByOrder?: SortByOrder,
 ) {
   switch (sortBy) {
     case 'Size':
-      return sortBySize(intersection, sortByOrder);
+      return sortBySize(intersections, sortByOrder);
     case 'Degree':
-      return sortByDegree(intersection, vSetSortBy, visibleSets, sortByOrder);
+      return sortByDegree(intersections, vSetSortBy, visibleSets, sortByOrder);
     case 'Deviation':
-      return sortByDeviation(intersection, sortByOrder);
+      return sortByDeviation(intersections, sortByOrder);
```

Comment:
The main reason these are separated is because the relevant sort variable is different for each. Definitely would be possible to combine these, but there'd be some significant logic in the function to differentiate.

Reply:
Shouldn't `sortByAttribute()` be able to sort by any attribute?

Reply:
Deviation and Degree are handled as attributes, but they don't share the same properties of every other attribute. That's why I have these handled as a edge cases separately. Attribute sort function could be generalized, but I'm not sure if it would add any additional readability. For degree especially, there's unique sorting behavior that involves visibleSets. This isn't touched by attribute sorting and doesn't really make sense to combine.